editor/plugins/texture_3d_editor_plugin.cpp

```diff
@@ -83,16 +83,16 @@ void Texture3DEditor::_notification(int p_what) {
 
 		case NOTIFICATION_DRAW: {
 			Ref<Texture2D> checkerboard = get_editor_theme_icon(SNAME("Checkerboard"));
-			Size2 size = get_size();
-
-			draw_texture_rect(checkerboard, Rect2(Point2(), size), true);
+			draw_texture_rect(checkerboard, texture_rect->get_rect(), true);
+			_draw_outline();
 		} break;
 
 		case NOTIFICATION_THEME_CHANGED: {
 			if (info) {
 				Ref<Font> metadata_label_font = get_theme_font(SNAME("expression"), EditorStringName(EditorFonts));
 				info->add_theme_font_override(SceneStringName(font), metadata_label_font);
 			}
+			cached_outline_color = get_theme_color(SNAME("extra_border_color_1"), EditorStringName(Editor));
```

Comment:
We tend to use a theme cache struct for things like this. Example in the header:

```cpp
	struct ThemeCache {
		Color valid_color;
		Color warning_color;
		Color error_color;
	} theme_cache;
```